src/server/updater/updatechecker.cpp

```diff
@@ -46,8 +45,69 @@ void UpdateChecker::setCallback(const std::function<void()> &callback) {
     _callback = callback;
 }
 
+const VersionInfo &UpdateChecker::versionInfo(const DistributionChannel channel) {
+    const VersionInfo &prodVersion = prodVersionInfo();
+
+    // If the user wants only `Production` versions, just return the current `Production` version.
+    if (channel == DistributionChannel::Prod) return prodVersion;
+
+    // Otherwise, we need to check if there is not a newer version in other channels.
+    /// The use wants `Beta` updates.
+    const VersionInfo &betaVersion =
+            _versionsInfo.contains(DistributionChannel::Beta) ? _versionsInfo[DistributionChannel::Beta] : _defaultVersionInfo;
+    if (channel == DistributionChannel::Beta) {
+        if (CommonUtility::isVersionLower(prodVersion.fullVersion(), betaVersion.fullVersion())) {
+            // Beta > Prod
+            return betaVersion;
+        }
+
+        // Prod > Beta
+        return prodVersion;
+    }
+
+    /// The user wants `Internal` updates.
+    const VersionInfo &internalVersion = _versionsInfo.contains(DistributionChannel::Internal)
+                                                 ? _versionsInfo[DistributionChannel::Internal]
+                                                 : _defaultVersionInfo;
+    if (channel == DistributionChannel::Internal) {
+        bool betaIsLower = false;
+        if (CommonUtility::isVersionLower(betaVersion.fullVersion(), internalVersion.fullVersion())) {
+            betaIsLower = true;
+        }
+        bool prodIsLower = false;
+        if (CommonUtility::isVersionLower(prodVersion.fullVersion(), internalVersion.fullVersion())) {
```

Comment:
We may sort a vector with the three versions in it and and return the latest version. The code does not say what to do in case of a tie (two channels with the same version).

Reply:
Done 50b1fc4bd083383bf8d52e77c5d48d68e810033a